processing/src/main/java/io/druid/segment/virtual/ExpressionObjectSelector.java

```diff
@@ -83,10 +83,7 @@ public Number get()
         // Unknown ValueType. Try making an Object selector and see if that gives us anything useful.
         final ObjectColumnSelector selector = columnSelectorFactory.makeObjectColumnSelector(columnName);
         final Class clazz = selector == null ? null : selector.classOfObject();
-        if (selector == null || (clazz != Object.class && Number.class.isAssignableFrom(clazz))) {
-          // We know there are no numbers here. Use a null supplier.
-          supplier = null;
-        } else {
+        if (selector != null && (clazz == Object.class || Number.class.isAssignableFrom(clazz))) {
```

Comment:
Could you please add a test for this condition? Also `clazz.isAssignableFrom(Number.class)` feels clearer than `clazz == Object.class`